src/test/rpc/NoRipple_test.cpp

```diff
@@ -222,42 +226,60 @@ class NoRipple_test : public beast::unit_test::suite
-        }
-        {
-            Json::Value params;
-            params[jss::account] = alice.human();
-            params[jss::peer] = gw.human();
+                auto lines =
+                    env.rpc("json", "account_lines", to_string(params));
+                auto const& line0 = lines[jss::result][jss::lines][0u];
+                BEAST_EXPECT(line0[jss::no_ripple_peer].asBool() == true);
+            }
+            {
+                params[jss::account] = alice.human();
+                params[jss::peer] = gw.human();
 
-            auto lines = env.rpc("json", "account_lines", to_string(params));
-            auto const& line0 = lines[jss::result][jss::lines][0u];
-            BEAST_EXPECT(line0[jss::no_ripple].asBool() == true);
-        }
-        {
-            Json::Value params;
-            params[jss::account] = gw.human();
-            params[jss::peer] = bob.human();
+                auto lines =
+                    env.rpc("json", "account_lines", to_string(params));
+                auto const& line0 = lines[jss::result][jss::lines][0u];
+                BEAST_EXPECT(line0[jss::no_ripple].asBool() == true);
+            }
+            {
+                params[jss::account] = gw.human();
+                params[jss::peer] = bob.human();
+
+                auto lines =
+                    env.rpc("json", "account_lines", to_string(params));
+                auto const& line0 = lines[jss::result][jss::lines][0u];
+                BEAST_EXPECT(line0[jss::no_ripple].asBool() == false);
+            }
+            {
+                params[jss::account] = bob.human();
+                params[jss::peer] = gw.human();
 
-            auto lines = env.rpc("json", "account_lines", to_string(params));
-            auto const& line0 = lines[jss::result][jss::lines][0u];
-            BEAST_EXPECT(line0[jss::no_ripple].asBool() == false);
+                auto lines =
+                    env.rpc("json", "account_lines", to_string(params));
+                auto const& line0 = lines[jss::result][jss::lines][0u];
+                BEAST_EXPECT(line0[jss::no_ripple_peer].asBool() == false);
+            }
         }
+        else
         {
-            Json::Value params;
-            params[jss::account] = bob.human();
-            params[jss::peer] = gw.human();
+            // test for transactions
+            {
+                params[jss::account] = bob.human();
+                params[jss::role] = "gateway";
+                params[jss::transactions] = "asdf";
 
-            auto lines = env.rpc("json", "account_lines", to_string(params));
-            auto const& line0 = lines[jss::result][jss::lines][0u];
-            BEAST_EXPECT(line0[jss::no_ripple_peer].asBool() == false);
+                auto lines =
+                    env.rpc("json", "noripple_check", to_string(params));
+                BEAST_EXPECT(lines[jss::result][jss::error] == "invalidParams");
```

Comment:
are we getting `invalidParams` due to the `jss::transactions` field? is that the cause of the problem?
I'm trying to understand how this testcase relates to the `noripple_check` property